x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/FieldValueFetcher.java

```diff
@@ -69,27 +71,43 @@ Object format(Object value) {
         }
     }
 
+    private static Map<String, FieldValueFetcher> build(SearchExecutionContext context, String field) {
+        MappedFieldType fieldType = context.getFieldType(field);
+        assert fieldType != null : "Unknown field type for field: [" + field + "]";
+        Map<String, FieldValueFetcher> fetchers = new HashMap<>();
+
+        if (fieldType instanceof AggregateDoubleMetricFieldMapper.AggregateDoubleMetricFieldType aggMetricFieldType) {
+            for (NumberFieldMapper.NumberFieldType metricSubField : aggMetricFieldType.getMetricFields().values()) {
+                if (context.fieldExistsInIndex(metricSubField.name())) {
+                    IndexFieldData<?> fieldData = context.getForField(metricSubField, MappedFieldType.FielddataOperation.SEARCH);
+                    fetchers.put(metricSubField.name(), new FieldValueFetcher(metricSubField.name(), fieldType, fieldData));
+                }
+            }
+        } else {
+            if (context.fieldExistsInIndex(field)) {
```

Comment:
Include only the fields that exist in the index, instead of all mapped field. 

This should improve downsampling performance for cases such as ECS schema.